src/main/java/http_server/Parser.java

```diff
@@ -0,0 +1,74 @@
+package http_server;
+
+import java.util.HashMap;
+
+public class Parser {
```

Comment:
This name might be too generic - `RequestParser` might be more precise

Reply:
It actually parses Requests and Responses (parsing Responses are used in tests)

Reply:
Interesting - I can't find where that's happening

Reply:
Here you go: https://github.com/serpient/java_http_server/blob/fd8f09979480f8c52de4e3362b8b2779a1abf745/src/test/java/http_server/ServerTest.java

The file difference was too large for it to show